Sources/SQLKit/Expressions/Clauses/SQLReturning.swift

```diff
@@ -1,30 +1,45 @@
-/// `RETURNING ...` statement.
+/// A clause describing a list of values to be returned from a data-modifying query.
+///
+/// Most - though not all - dialects support `RETURNING` clauses for DML queries such as `INSERT`, `UPDATE`, and
+/// `DELETE`. This clause will not emit any SQL if the dialect reports lacking this support.
+///
+/// This clause is the building block underlying ``SQLReturningBuilder``.
 public struct SQLReturning: SQLExpression {
+    /// The list of columns to be returned.
+    ///
+    /// If empty, the expression does not serialize any content.
     public var columns: [any SQLExpression]
 
-    /// Creates a new `SQLReturning`.
+    /// Create a new returning-values clause.
+    ///
+    /// - Parameter column: A single column to return from a query.
     @inlinable
     public init(_ column: SQLColumn) {
         self.init([column])
     }
 
-    /// Creates a new `SQLReturning`.
+    /// Creates a new returning-values clause.
+    ///
+    /// - Parameter columns: One or more columns to return from a query.
     @inlinable
     public init(_ columns: [any SQLExpression]) {
         self.columns = columns
     }
 
+    // See `SQLExpression.serialize(to:)`.
     public func serialize(to serializer: inout SQLSerializer) {
         guard serializer.dialect.supportsReturning else {
-            serializer.database.logger.warning("\(serializer.dialect.name) does not support 'RETURNING' clause, skipping.")
+            /// This logging can be a bit noisy for MySQL users, elide it for now.
+            //serializer.database.logger.warning("\(serializer.dialect.name) does not support 'RETURNING' clause, skipping.")
```

Comment:
I have been thoroughly inconsistent in my handling of places where there were assertions, logs, etc. in the existing code, but I've tried to tend towards silencing things completely (since in the present day it generally isn't useful to have the warnings anyway). The much more correct solution would be for serialization to be able to throw so that obvious syntax problems can be reported properly, but of course that requires the upcoming major version break.

Reply:
How about an assertion to trigger in debug and then a log at debug as per the log level guidelines?